rtos/TARGET_CORTEX/TOOLCHAIN_ARM_STD/mbed_boot_arm_std.c

```diff
@@ -27,18 +27,28 @@
 __value_in_regs struct __argc_argv __rt_lib_init(unsigned heapbase, unsigned heaptop);
 void _platform_post_stackheap_init(void);
 
-#if !defined(ISR_STACK_SIZE)
-extern uint32_t               Image$$ARM_LIB_STACK$$ZI$$Base[];
-extern uint32_t               Image$$ARM_LIB_STACK$$ZI$$Length[];
-#define ISR_STACK_START       ((unsigned char*)Image$$ARM_LIB_STACK$$ZI$$Base)
-#define ISR_STACK_SIZE        ((uint32_t)Image$$ARM_LIB_STACK$$ZI$$Length)
+extern uint32_t Image$$ARM_LIB_STACK$$ZI$$Base[];
+extern uint32_t Image$$ARM_LIB_STACK$$ZI$$Length[];
+
+extern uint32_t Image$$RW_IRAM1$$ZI$$Limit[];
+
+extern uint32_t Image$$ARM_LIB_HEAP$$ZI$$Base[];
```

Comment:
Currently `ARM_LIB_HEAP` is specified only for few targets (`ARM_LIB_STACK` for all), this means that the following change will cause build failure for most of the targets.

To fully move to the 2 region memory model we need to specify `ARM_LIB_HEAP` in all scatter files and `__HeapLimit` for GCC linker scripts. Then we will be able to completely remove `ISR_STACK_SIZE, ISR_STACK_START, ISR_HEAP_SIZE, ISR_HEAP_START` macros and relay only on the linker script symbols to define heap and stack.

I already tried that, but I took a step back since there was too much changes and we have discussed this here (and few next posts):
https://github.com/ARMmbed/mbed-os/pull/9092#issuecomment-448616630

Reply:
Agree on __HeapLimit, will be needed for all GCC linker files, for now exploring the options to not have ARM_LIB_HEAP in all targets, moving it to mbed_rtx.h back seems reasonable.

Reply:
Also I will not remove existing symbols for HEAP and Stack, since we have few targets who use completely different ram bank for heap, or want to expand heap to multiple banks. Symbols will be needed for those cases.

Reply:
Added __HeapLimit for all targets in GCC linker scripts.